python/ray/serve/http_proxy.py

```diff
@@ -147,6 +273,22 @@ async def _not_found(self, scope, receive, send):
             status_code=404)
         await response.send(scope, receive, send)
 
+    async def _fallback_to_prefix_router(self, scope, receive, send):
+        route_prefix, handle = self.prefix_router.match_route(
+            scope["path"], scope["method"])
+        if route_prefix is None:
+            return await self._not_found(scope, receive, send)
+
+        # Modify the path and root path so that reverse lookups and redirection
+        # work as expected. We do this here instead of in replicas so it can be
+        # changed without restarting the replicas.
+        if route_prefix != "/":
+            assert not route_prefix.endswith("/")
+            scope["path"] = scope["path"].replace(route_prefix, "", 1)
+            scope["root_path"] = route_prefix
```

Comment:
Should this be covered by a test that does reverse lookups and redirection (or is there already such a test?)

Reply:
Ah I think I wrote this comment. You can add a test for `url_path_for` https://stackoverflow.com/questions/63682956/fastapi-retrieve-url-from-view-name-route-name for something...

Reply:
Adding a test for this in a separate PR because it's unrelated https://github.com/ray-project/ray/pull/15461